readme.md

```diff
@@ -43,33 +43,93 @@ Insert text at the cursor, replacing any possible selected text:
 const textarea = document.querySelector('textarea');
 const button = document.querySelector('.js-add-signature');
 button.addEventListener(event => {
-	insertText(textarea, 'Made by 🐝 with pollen.');
+	textFieldEdit.insert(textarea, 'Made by 🐝 with pollen.');
 });
 ```
 
-This will wrap the selected text (if any) with `**` on both sides:
+This will act like `field.value = 'New value'` but with **undo** support and by firing the `input` event:
 
 ```js
 const textarea = document.querySelector('textarea');
-const button = document.querySelector('.js-markdown-make-text-bold');
-button.addEventListener(event => {
-	// Don't use `getSelection()` if you want Firefox support
-	const selectedText = value.slice(
-		textarea.selectionStart,
-		textarea.selectionEnd
-	);
-	insertText(textarea, '**' + selectedText + '**');
+const resetButton = document.querySelector('.js-markdown-reset-field');
+resetButton.addEventListener(event => {
+	textFieldEdit.set(textarea, 'New value');
 });
 ```
 
-This will replace the entire content, equivalent to `field.value = 'New text!'` but with **undo** support:
+## API
+
+### textFieldEdit.insert(field, text)
+
+Inserts `text` at the cursor’s position, replacing any selection.
+
+```js
+const field = document.querySelector('input[type="text"]');
+textFieldEdit.insert(field, '🥳');
+// Changes field's value from 'Party|' to 'Party🥳|' (where | is the cursor)
+```
+
+#### field
+
+Type: `HTMLTextAreaElement` `HTMLInputElement`
+
+#### text
+
+Type: `string`
+
+The text to insert at the cursor's position.
+
+### textFieldEdit.set(field, text)
+
+Replaces the entire content, equivalent to `field.value = 'New text!'` but with **undo** support and by firing the `input` event:
 
 ```js
 const textarea = document.querySelector('textarea');
-textarea.select(); // The text needs to be selected so it will be replaced
-insertText(textarea, 'New text!');
+textFieldEdit.set(textarea, 'New text!');
+```
+
+#### field
+
+Type: `HTMLTextAreaElement` `HTMLInputElement`
```

Comment:
FYI, I started doing this instead:

```suggestion
Type: `HTMLTextAreaElement | HTMLInputElement`
```

I kinda like how I can just copy-paste the TS types into the readme. Makes it easier to keep them in sync.